java/vector/src/main/java/org/apache/arrow/vector/util/Validator.java

```diff
@@ -117,7 +117,7 @@ public static void compareVectorSchemaRoot(VectorSchemaRoot root1, VectorSchemaR
    */
   public static void compareFieldVectors(FieldVector vector1, FieldVector vector2) {
     Field field1 = vector1.getField();
-    if (!field1.equals(vector2.getField())) {
+    if (vector1.getClass() != vector2.getClass()) {
```

Comment:
Why change this?

Reply:
Field implemented hashCode but it also compare names, that is to say, two same type vector with different name will not equal here even they have same elements, so compare with getClass is better?

Reply:
We need to compare Field names for the integration tests. Ideally, we should be able to compare just the data in `ValueVector`s and the entire `FieldVector` also. For the dictionary vector, I don't think the field name is important, but shouldn't hurt to include the name check as well. Were you seeing a test failure if you revert this?

Reply:
Like https://github.com/apache/arrow/pull/4843/files#diff-945722295a5f95e681897e48eedfcb08R459 vectors with different names ("v1" "v2") will not equals.
If you think we should change behavior of Validator, maybe we could refactor this PR and do not use Validator logic after https://github.com/apache/arrow/pull/4933 merged, that PR introduces equals API in ValueVector. What do you think and could you also take a look? thanks!

Reply:
I would rather not change the behavior of Validator here, can you revert this and we can make it less restrictive later?

Reply:
Sure, fixed.